### DISCUSSION TITLE
deploy node has this err

### DISCUSSION THREAD

**Discussion by malingzhao** [zkSync-Community-Hub/zksync-developers#678] [General] [answered]
git clone https://github.com/matter-labs/zksync-era.git
cd zksync-era/docs/guides/external-node
cd docker-compose-examples
docker compose --file mainnet-external-node-docker-compose.yml up

2024-09-04T06:31:15.463957Z  INFO NamedFuture{name=node_storage_initializer}: zksync_node_framework::implementations::layers::node_storage_init: Node storage initialization task completed
Error: One or more tasks failed: [TaskFailed(TaskId("oneshot_runner"), Oneshot task node_storage_initializer_precondition failed

Caused by:
    0: Failed acquiring DB connection [requested by `en` at core/node/node_storage_init/src/external_node/snapshot_recovery.rs:75]: pool timed out while waiting for an open connection

### Repo Link (Optional)

https://github.com/matter-labs/zksync-era

### Additional Details

docker logs -f docker-compose-examples-external-node-1
2024-09-04T06:31:15.463957Z  INFO NamedFuture{name=node_storage_initializer}: zksync_node_framework::implementations::layers::node_storage_init: Node storage initialization task completed
Error: One or more tasks failed: [TaskFailed(TaskId("oneshot_runner"), Oneshot task node_storage_initializer_precondition failed

Caused by:
    0: Failed acquiring DB connection [requested by `en` at core/node/node_storage_init/src/external_node/snapshot_recovery.rs:75]: pool timed out while waiting for an open connection

**Comment by alhenaai** [comment]
It appears that the error you're encountering is related to a timeout issue while trying to acquire a database connection during the node storage initialization process. This can happen if the database connection pool is exhausted or if the database is not responding in a timely manner.

Here are a few steps you can take to resolve this issue:

1. **Increase Database Connection Pool Size**: Check your database configuration and increase the number of connections allowed in the pool. This setting is often found in the database configuration file or environment variables related to your deployment setup.

2. **Optimize Database Performance**: Ensure that your database is optimized for performance. This includes regular maintenance tasks such as indexing, vacuuming (for PostgreSQL), and checking for slow queries that might be locking up resources.

3. **Check Network Latency**: If your database is hosted separately from your application, network latency could be causing timeouts. Consider improving network conditions or moving the database closer to your application server.

4. **Resource Allocation**: Ensure that your Docker containers have sufficient resources (CPU, memory) allocated. Resource constraints can lead to slow performance and timeouts.

5. **Database Logs**: Review the database logs for any warnings or errors that could indicate why connections are being held longer than expected or why the connection pool is exhausted.

If the issue persists after trying these steps, you might want to look into the specific configurations of your ZKsync node setup or consult with a database administrator if the problem seems to be related to the database itself.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by malingzhao** [reply]
docekr logs -f docker-compose-examples-postgres-1
2024-09-04 06:34:44.170 UTC [68864] LOG:  unexpected EOF on client connection with an open transaction

**Reply by alhenaai** [reply]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Reply by bxpana** [reply]
@malingzhao could you provide some more details on where you are getting the issue? Which step are you running in the [repo](https://github.com/matter-labs/zksync-era) to get this?

**Comment by malingzhao** [comment]
git clone https://github.com/matter-labs/zksync-era.git
cd zksync-era/docs/guides/external-node
cd docker-compose-examples
docker compose --file mainnet-external-node-docker-compose.yml up
the latest version   then  2 hours later  has this err

**Reply by bxpana** [reply]
I see you're following the instructions [here](https://docs.zksync.io/zksync-node/quickstart). 

You're trying to deploy a mainnet external node, did you confirm your system meets the [requirements](https://docs.zksync.io/zksync-node/quickstart#system-requirements)? CPU: A relatively modern CPU is recommended.
RAM: 32 GB
Storage:
Mainnet Nodes: 300 GB, with the state growing about 1TB per month.
Network: 100 Mbps connection (1 Gbps+ recommended)

**Comment by malingzhao** [comment]
the latest version of the zksync
    image: "matterlabs/external-node:2.0-v24.15.0" is ok
        image: "matterlabs/external-node:2.0-v24.16.0"  has this err

**Reply by bxpana** [reply]
Checking with the team to see if anyone has run into this yet.

**Reply by bxpana** [reply]
@malingzhao we have had another report of this issue and the team is looking into it for now.

**Reply by bxpana** [reply, marked as answer]
The v24.25.0 release should resolve this issue and should be released in the next week or so.
